### ISSUE TITLE
Client Provider Failure: Error converting value "FIELD_DEFINITION"

### ISSUE THREAD

**Issue by NatElkins** [fsprojects/FSharp.Data.GraphQL#224]
### Description

When trying to hit a local Apollo GraphQL server, the type provider fails.

### Repro steps

Unfortunately I'm operating against a closed source GraphQL server.  But what I did was create a console app, reference `nuget FSharp.Data.GraphQL.Client 1.0.0-beta5`, and then try and use it like so:

```fsharp
type A = GraphQLProvider<"http://localhost:4000">
```

### Expected behavior

It shouldn't fail.

### Actual behavior

When just pointing to the GraphQL server without the correct route:

<img width="877" alt="image" src="https://user-images.githubusercontent.com/1795462/57782525-65015b00-76fa-11e9-8706-8ee1de31f2f1.png">

When pointing to the GraphQL path:

<img width="1195" alt="image" src="https://user-images.githubusercontent.com/1795462/57782262-ddb3e780-76f9-11e9-94a8-b101c8665a1a.png">

### Known workarounds

Nada.

### Related information 

* Operating system - macOS
* .NET Runtime, CoreCLR or Mono Version - .NET Core 2.2.103
* Using FSharp.Data.GraphQL.Client version 1.0.0-beta5 .

**Comment by NatElkins** [comment]
Also, oddly when I build this the error I see is

```
/Users/nat/Projects/rr2/ReproExec/Program.fs(4,10): error FS1108: The type 'String' is required here and is unavailable. You must add a reference to assembly 'System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e'. [/Users/nat/Projects/rr2/ReproExec/ReproExec.fsproj]
```
 but that might be a red herring.

**Comment by johnberzy-bazinga** [comment]
@NatElkins Are you trying the provider in FSI?

**Comment by NatElkins** [comment]
Nope.

**Comment by johnberzy-bazinga** [comment]
Do you think you can share the introspection schema Json? Also, are you using a recent version of Apollo?

**Comment by NatElkins** [comment]
1. Sure
[schema.txt](https://github.com/fsprojects/FSharp.Data.GraphQL/files/3183782/schema.txt)

2. I think so.  I'm looking at the package-lock.json and I see `apollo-engine` has a version of 1.1.2.  https://www.npmjs.com/package/apollo-engine .

**Comment by johnberzy-bazinga** [comment]
![Screen Shot 2019-05-15 at 10 49 21 AM](https://user-images.githubusercontent.com/18220364/57797394-346ff000-76ff-11e9-9765-6eb755095a21.png)

Strange. It seems to work (at least in FSI). At least we can rule out the introspection schema. Are you trying in a console or library project?

**Comment by johnberzy-bazinga** [comment]
Also, if you're using VS Code, are you using the net or netcore language service?

Thanks

**Comment by NatElkins** [comment]
Console project.

Here's what I get when I try to use Ionide (I was using Rider before).

<img width="877" alt="image" src="https://user-images.githubusercontent.com/1795462/57797836-5aa28980-7719-11e9-99ba-83ff7b608b31.png">

**Comment by NatElkins** [comment]
Library works:

<img width="877" alt="image" src="https://user-images.githubusercontent.com/1795462/57798062-cf75c380-7719-11e9-8af9-82987c6fb974.png">

**Comment by johnberzy-bazinga** [comment]
@NatElkins Thanks for debugging. So I guess it's an issue with netcoreapp at designtime. Are you using netcoreapp2.0?

**Comment by NatElkins** [comment]
Super weird, after I got it working in the library it started working everywhere (both Rider and Ionide, console and library).

I was using netcoreapp2.2.  I just flipped it to 2.1 and then back to 2.2...can't quite remember the order I did things.

**Comment by johnberzy-bazinga** [comment]
Super strange. We did notice some weird behaviour too, but that was only after updating packages. We attributed the issue to some sort of caching.

**Comment by NatElkins** [comment]
Actually I think I spoke to soon.  I often get a "Critical Failure" error with Rider, which kills all Intellisense.  Could be their issue though.  Seems like I need to get the right order of having the GraphQL service be up and console/library being in a cleaned and restored state.

Also seeing some things like this:
<img width="877" alt="image" src="https://user-images.githubusercontent.com/1795462/57799283-b9b5cd80-771c-11e9-89cd-b4ab95ae6f13.png">

**Comment by johnberzy-bazinga** [comment]
@NatElkins strange. We don't use rider, but I'll give it a try when I have some time.

**Comment by johnberzy-bazinga** [comment]
@NatElkins Do you mind trying with beta-07 to see if the crashing behaviour has been improved?

Thanks,
John

**Comment by NatElkins** [comment]
Still getting the same behavior when I point it to a locally running server with Rider, but it works when pointing to a local server using Ionide.

I noticed that Rider runs its Intellisense using Mono, while I have Ionide set to use `netcore`.  When I change Ionide to run using `net`, it doesn't seem to work as well and stops producing Intellisense when I navigate to my `Query` type (can be seen in the previous image I posted).

In summary...a little improvement when using from Ionide, no improvement when using from Rider.

**Comment by johnberzy-bazinga** [comment]
@NatElkins That's strange. We're using the typeprovider in a Fabulous app with VS for Mac. If I'm not mistaken, VS for mac also runs intellisense using Mono. It may be an older version of Mono, though. I think 5.18.x.

Do you know which version of Mono you're using?

**Comment by NatElkins** [comment]
Mono JIT compiler version 5.18.1.3 (2018-08/fdb26b0a445 Wed Mar 20 10:02:02 EDT 2019)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:
	SIGSEGV:       altstack
	Notification:  kqueue
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug
	Interpreter:   yes
	LLVM:          yes(600)
	Suspend:       preemptive
	GC:            sgen (concurrent by default)

**Comment by NatElkins** [comment]
Can't seem to repro this issue anymore with version 1.0.2.
